### PR TITLE
fix(aio): fix window title for Home page

### LINKED ISSUE #20427: aio has title on the homepage set to "Angular - link" instead of just "Angular"
if you navigate to [angular.io](https://angular.io), you'll see:

screen shot 2017-11-14 at 5 15 11 pm

## Brief analysis of how this happens:

When we generate `marketing/index.json` for the first page we turn the original `marketing/index.md` code:

```
<article>
  <h1 class="no-toc" style="display: none"></h1>
  ...
</article>
```

into:

```
<article>
  <h1 class=\"no-toc\" style=\"display: none\" id=\"\"><a title=\"Link to this heading\" class=\"header-link\" aria-hidden=\"true\" href=\"index#\"><i class=\"material-icons\">link</i></a></h1>
  ...
</article>
```

which then gets picked up by doc-viewer which then finds the "link" text node and uses that to incorrectly set the title: https://github.com/angular/angular/blob/24cf8b326918e0694dfcdcf48635b5325060827a/aio/src/app/layout/doc-viewer/doc-viewer.component.ts#L99-L106

### PR DESCRIPTION
**Pull request by gkalpak**
## PR Checklist
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] ~~Docs have been added / updated (for bug fixes / features)~~

## PR Type
```
[x] Bugfix
```

## What is the current behavior?
`Angular - link` appears as the window title on the angular.io Home page.
Issue Number: #20427

## What is the new behavior?
`Angular` appears as the window title on the angular.io Home page.
See #20427 and the commit messages for more details.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
Fixes #20427.

### CODE REVIEW

**Comment by gkalpak** [comment]
The first commit fixes the issue by replacing `display: none` with `visibility: hidden` on the `<h1>`.
The second commit proposes an alternative fix by not generating anchors for `.no-anchor` headings.

I prefer the second approach. Left them both for reference. I'll squash the commits before merging if we decide to go with the second one.

**Comment by gkalpak** [comment]
The CircleCI linting error is not related to this PR (comes from master).

File: aio/tools/transforms/angular-base-package/post-processors/autolink-headings.js [hunk 1]
```diff
@@ -12,11 +37,13 @@ module.exports = [
...
-      children: [{ type: 'text', value: 'link' }]
-    }
+    content: [
+      {
+        type: 'element',
+        tagName: 'i',
+        properties: {className: ['material-icons']},
+        children: [{ type: 'text', value: 'link' }]
+      }
+    ]
```
Comment (petebacondarwin) [inline aio/tools/transforms/angular-base-package/post-processors/autolink-headings.js]: Should content be an array? Was this a mistake in the original on my part?

Comment (petebacondarwin) [reply aio/tools/transforms/angular-base-package/post-processors/autolink-headings.js]: Oh I see that in woorm's version he wraps the content in an array.

**Review by petebacondarwin — APPROVED** [review]
LGTM - let's go with option 2 (and squash)

**Pull request closed without merging**